// @vuepress/core/lib/app/createAppPages.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAppPages = void 0;
const utils_1 = require("@vuepress/utils");
const page_1 = require("../page");
const log = utils_1.debug('vuepress:core/app');
/**
 * Create pages for vuepress app
 */
const createAppPages = async (app) => {
    log('createAppPages start');
    // resolve page absolute file paths according to the page patterns
    const pageFilePaths = await utils_1.globby(app.options.pagePatterns, {
        absolute: true,
        cwd: app.dir.source(),
    });
    // create pages from files
    const pages = await Promise.all(pageFilePaths.map((filePath) => page_1.createPage(app, { filePath })));
    // if there is no 404 page, add one
    if (!pages.some((page) => page.path === '/404.html')) {
        pages.push(await page_1.createPage(app, {
            path: '/404.html',
            frontmatter: {
                layout: '404',
            },
        }));
    }
    log('createAppPages finish');
    return pages;
};
exports.createAppPages = createAppPages;
