// @vuepress-reco/style-default/lib/node/tailwind/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.tailwindConfig = void 0;
const plugin_md_style_1 = require("./plugin-md-style");
const custom_colors_1 = require("./custom-colors");
// const colors = require('tailwindcss/colors')
exports.tailwindConfig = {
    purge: ['./src/**/*.vue'],
    darkMode: 'media',
    theme: {
        extend: {
            backgroundOpacity: {
                98: '.98'
            },
            borderRadius: {
                '1/2': '50%',
            },
            borderWidth: {
                6: '6px',
            },
            // boxShadow 不支持嵌套
            boxShadow: {
                light: '0 1px 8px 0 rgba(0, 0, 0, 0.1)',
                'light-heavier': '0 2px 16px 0 rgba(0, 0, 0, 0.2)',
                dark: '0 1px 8px 0 rgba(0, 0, 0, .6)',
                'dark-heavier': '0 2px 16px 0 rgba(0, 0, 0, .7)'
            },
            colors: custom_colors_1.colors,
            height: () => ({ 'screen-3/5': '60vh' }),
            zIndex: {
                'negative-10': -10
            }
        },
    },
    variants: {
        extend: {
            borderColor: ['dark'],
            borderWidth: ['dark'],
            boxShadow: ['dark'],
        },
    },
    plugins: [plugin_md_style_1.mdStylePlugin],
};
